File: sample/javascript/mainAnimateNodeSample.mjs
import BuildBox from './buildBox.mjs';

(async () => {
  const roomName = '1000';
  const buildBox = new BuildBox(roomName);

  buildBox.setBoxSize(0.5);
  buildBox.setBuildInterval(0.01);

  for (let i = 0; i < 10; i++) {
    buildBox.createBox(-1, i, 0, 0, 1, 1);
    buildBox.createBox(0, i, 0, 1, 0, 0);
    buildBox.createBox(1, i, 0, 1, 1, 0);
    buildBox.createBox(2, i, 0, 0, 1, 1);
  }

  for (let i = 0; i < 5; i++) {
    buildBox.removeBox(0, i * 2, 0);
    buildBox.removeBox(1, i * 2 + 1, 0);
  }

  await buildBox.sendData();

  await buildBox.sleepSecond(1)

  buildBox.animateNode(10, 0, 0, 0, 30, 0, 2, 10)
  buildBox.sendData()

})().catch(error => {
  console.error(error);
});

File: sample/python/main_animate_node_sample.py
from time import sleep
# voxelammingパッケージからVoxelammingクラスをインポートします
from voxelamming_local import Voxelamming

# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# Voxelammingクラスのインスタンスを生成します
voxelamming = Voxelamming(room_name)
# ボクセルの設定を行います
voxelamming.set_box_size(0.5)
voxelamming.set_build_interval(0.01)

for i in range(10):
    voxelamming.create_box(-1, i, 0, r=0, g=1, b=1)
    voxelamming.create_box(0, i, 0, r=1, g=0, b=0)
    voxelamming.create_box(1, i, 0, r=1, g=1, b=0)
    voxelamming.create_box(2, i, 0, r=0, g=1, b=1)

for i in range(5):
    voxelamming.remove_box(0, i * 2 + 1, 0)
    voxelamming.remove_box(1, i * 2, 0)

# ボクセルデータをアプリに送信します。（1回目）
voxelamming.send_data()

# 1秒待機します
sleep(0.1)

voxelamming.animate(10, 0, 0, pitch=0, yaw=30, roll=0, scale=2, interval=10)

# ボクセルデータをアプリに送信します。（2回目）
voxelamming.send_data()
